core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -158,14 +158,22 @@ String getPreferredIp(List<String> preferredTypes) {
    */
   void forceRefresh() {
     synchronized (instanceDataGuard) {
-      // If a scheduled refresh hasn't started, perform one immediately
-      if (nextInstanceData.cancel(false)) {
+      String logMessage;
+      nextInstanceData.cancel(false);
+      if (nextInstanceData.isCancelled()) {
+        logMessage =
```

Comment:
Why not just write the log messages instead of setting a variable? For me, logging what happens as it happens is easier to read than setting a variable and logging it later.